### PR TITLE
Brand synchronization 

### PR DESCRIPTION
**Pull request by cwhite911**
Synchronizing GRASS brand with Quarto

## Updated PR Details

I've added a style guide partially as a way to check edge cases. Set it up locally you need to add an `_environment.local` file to the root of the project.

The file should include the following

```bash
PYTHONDEVMODE=1
QUARTO_PROFILE=local
``` 

Now when you run `quarto preview` the content in `content/tests` will render with the project.

```bash
quarto preview
```
Then navigate to local site at `content/tests/styling.html`

`http://localhost:5753/content/tests/styling.html`

> **Remember to update your port number**

## GRASS Quarto Design Guide

# Orginial Info
> I've added the `_brand.yml` file for us to use once we update Quarto to >=1.7

The formatting is close but still requires some additional tweaks.

- [https://posit-dev.github.io/brand-yml/](https://posit-dev.github.io/brand-yml/)
- [https://quarto.org/docs/advanced/typst/brand-yaml.html](https://quarto.org/docs/advanced/typst/brand-yaml.html)

### CODE REVIEW

**Comment by cwhite911** [comment]
image

**Comment by cwhite911** [comment]
image

**Comment by veroandreo** [comment]
I liked the previous font better or something similar to that sort of simpler letters (I do not know how they are called); this one does not seem to align with GRASS website style nor with the font in our logo.

**Comment by cwhite911** [comment]
> I liked the previous font better or something similar to that sort of simpler letters (I do not know how they are called); this one does not seem to align with GRASS website style nor with the font in our logo.

The font issues was a bug in the config should be fixed now. Will post an update screenshot when I'm at my computer.

**Comment by cwhite911** [comment]
I've added the start to a style guide partially as a way to check edge cases. Set it up locally you need to add an `_environment.local` file to the root of the project.

The file should include the following

```bash
PYTHONDEVMODE=1
QUARTO_PROFILE=local
``` 

Now when you run `quarto preview` the content in `content/tests` will render with the project.

```bash
quarto preview
```
Then navigate to local site at `content/tests/styling.html`

`http://localhost:5753/content/tests/styling.html`

> **Remember to update your port number**

## GRASS Quarto Design Guide

Screenshot 2025-05-06 at 22-16-27 GRASS Quarto Design Guide

**Comment by cwhite911** [comment]
Here are some more screenshots.

Please ignore *Team Documents* and *GRASS Quarto Design Guide* in the All Tutorials section. They are only there because of the local environmental variables from `_environment.local`. The `"missing"` images seem to be an artifact from my screenshot as they are present on in the browser.

## index

image

## Getting started with GRASS
image

**Comment by cwhite911** [comment]
Updated view:

Screenshot 2025-05-07 at 19-57-07 GRASS Quarto Design Guide

**Comment by cwhite911** [comment]
@petrasovaa please review when you are able.

Here are screenshots from the updated sections.

## When title-block-banner: true
Screenshot from 2025-05-08 13-42-36

## Buttons

Screenshot from 2025-05-08 13-42-54

## Modified Callouts
Screenshot from 2025-05-08 13-43-22

File: _freeze/content/tests/styling/execute-results/html.json
Comment (petrasovaa) [inline _freeze/content/tests/styling/execute-results/html.json:1]: I removed the freeze, it's being built now in CI

File: _variables.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+grass:
+  name: GRASS
+  version: 8.4
```
Comment (petrasovaa) [inline _variables.yml:3]: Why version? We would need to keep it updated. I would remove it.

File: content/tests/listing.qmd [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Team Documents
+description: Learn about sytling your GRASS tutorial
```
Comment (petrasovaa) [inline content/tests/listing.qmd:3]:
```suggestion
description: Learn about styling your GRASS tutorial
```

File: content/tests/listing.qmd
Comment (petrasovaa) [inline content/tests/listing.qmd:1]: What is this for?

File: content/tests/styling.qmd
Comment (petrasovaa) [inline content/tests/styling.qmd:1]: This could be useful, but it shows up as one of the tutorials, which is confusing. Also it doesn't include the language content tabs, which we use in some tutorials and should be probably demonstrated here?

File: content/tests/styling.qmd [hunk 3]
```diff
@@ -0,0 +1,932 @@
+---
+title: GRASS Quarto Design Guide
+description: Adopted from Quarto Documentatoin
```
Comment (petrasovaa) [inline content/tests/styling.qmd:3]:
```suggestion
description: Adopted from Quarto Documentation
```

**Review by petrasovaa — CHANGES_REQUESTED** [review]
Looks good! I am not sure I understand the plan with the Team Documents and the guide.

Also, I thought we decided on sentence case for headings, no?

File: content/tests/listing.qmd
Comment (cwhite911) [reply content/tests/listing.qmd:1]: It is used to show what the different listing styles look like in styling.qmd

File: content/tests/styling.qmd
Comment (cwhite911) [reply content/tests/styling.qmd:1]: Did you do this part from the PR?

------
I've added a style guide partially as a way to check edge cases. Set it up locally you need to add an `_environment.local` file to the root of the project.

The file should include the following

```bash
PYTHONDEVMODE=1
QUARTO_PROFILE=local
``` 

Now when you run `quarto preview` the content in `content/tests` will render with the project.

```bash
quarto preview
```
Then navigate to local site at `content/tests/styling.html`

`http://localhost:5753/content/tests/styling.html`

> **Remember to update your port number**

File: _variables.yml (hunk 1 above)
Comment (cwhite911) [reply _variables.yml:3]: I thought it would be useful to have the latest version as a variable so if it is mentioned we can do a quick update.

File: _freeze/content/tests/styling/execute-results/html.json
Comment (cwhite911) [reply _freeze/content/tests/styling/execute-results/html.json:1]: Ok, we should add `_freeze/` to .gitignore then.

**Review by cwhite911 — COMMENTED** [review]
@petrasovaa thanks for the feedback. I've addressed your comments. My todo is to remove the _freeze files, and I'll add a section to show the language selection tabs.

File: content/tests/styling.qmd [hunk 4]
```diff
@@ -0,0 +1,932 @@
...
+
+    Continued (indent 4 spaces)
+
+```md
+*   item 2
+
+    Continued (indent 4 spaces)
+```
+
+(@)  A list whose numbering
```
Comment (veroandreo) [inline content/tests/styling.qmd:141]:
```suggestion
(@)  A list which numbering
```

File: content/tests/styling.qmd [hunk 5]
```diff
@@ -0,0 +1,932 @@
...
+
+continues after
+
+(@)  an interruption
+
+```md
+(@)  A list whose numbering
```
Comment (veroandreo) [inline content/tests/styling.qmd:148]:
```suggestion
(@)  A list which numbering
```

File: content/tests/styling.qmd [hunk 6]
```diff
@@ -0,0 +1,932 @@
...
+[^longnote]: Here's one with multiple blocks.
+
+    Subsequent paragraphs are indented to show that they
+belong to the previous footnote.
+
+        { some.code }
+
+    The whole paragraph can be indented, or just the first
+    line.  In this way, multi-paragraph footnotes work like
+    multi-paragraph list items.
```
Comment (veroandreo) [inline content/tests/styling.qmd:204]:
```suggestion
    line. In this way, multi-paragraph footnotes work like
    multi-paragraph list items.
```

File: content/tests/styling.qmd [hunk 7]
```diff
@@ -0,0 +1,932 @@
...
+
+[^longnote]: Here's one with multiple blocks.
+
+    Subsequent paragraphs are indented to show that they
+belong to the previous footnote.
+
+        { some.code }
+
+    The whole paragraph can be indented, or just the first
+    line.  In this way, multi-paragraph footnotes work like
```
Comment (veroandreo) [inline content/tests/styling.qmd:226]:
```suggestion
    line. In this way, multi-paragraph footnotes work like
```

File: content/tests/styling.qmd [hunk 8]
```diff
@@ -0,0 +1,932 @@
...
+```md
+::: {.callout-warning}
+This is a warning callout.
+:::
+```
+
+### Important
+
+::: {.callout-important}
+This is a imporant callout.
```
Comment (veroandreo) [inline content/tests/styling.qmd:551]:
```suggestion
This is an important callout.
```

File: content/tests/styling.qmd [hunk 9]
```diff
@@ -0,0 +1,932 @@
...
+:::
+
+```md
+::: {.callout-important}
+This is a imporant callout.
```
Comment (veroandreo) [inline content/tests/styling.qmd:556]:
```suggestion
This is an important callout.
```

File: content/tests/styling.qmd [hunk 10]
```diff
@@ -0,0 +1,932 @@
...
+[Warning Outline](#){.btn .btn-outline-warning .btn role="button"}
+[Info Outline](#){.btn .btn-outline-info .btn role="button"}
+[Light Outline](#){.btn .btn-outline-light .btn role="button"}
+[Dark Outline](#){.btn .btn-outline-dark .btn role="button"}
+[Link Outline](#){.btn .btn-outline-link .btn role="button"}
+```
+
+### Special Buttons
+
+We have a custom support buttons ([More Info](#includes)) that link to GRASS's [Open Collective page]({{< var grass.support >}})
```
Comment (veroandreo) [inline content/tests/styling.qmd:652]:
```suggestion
We have custom support buttons ([More Info](#includes)) that link to GRASS's [Open Collective page]({{< var grass.support >}})
```

File: content/tests/styling.qmd [hunk 11]
```diff
@@ -0,0 +1,932 @@
...
+
+page 1
+
+{{< pagebreak >}}
+
+page 2
+
+## Divs & Spans
+
+### Boarder
```
Comment (veroandreo) [inline content/tests/styling.qmd:793]:
```suggestion
### Border
```

File: content/tests/styling.qmd [hunk 12]
```diff
@@ -0,0 +1,932 @@
...
+
+More content.
+:::::
+```
+
+### Spans
+
+[This is *some text*]{.class key="val"}
+
+<span class="custom-span">This is a custom span.</span>
```
Comment (veroandreo) [inline content/tests/styling.qmd:831]: These 2 do not render anything special... is that right?

File: _variables.yml (hunk 1 above)
Comment (petrasovaa) [reply _variables.yml:3]: Let's remove it.

File: content/tests/styling.qmd [hunk 13]
```diff
@@ -0,0 +1,932 @@
...
+:::
+
+### Column Full screen shaded inset
+::: {.column-shaded-screen-inset}
+![A full screen image](images/LinkedIn-Post-Image.png)
+:::
+
+### Column Margin
+::: {.column-margin}
+![A image in the margin](images/LinkedIn-Post-Image.png)
```
Comment (veroandreo) [inline content/tests/styling.qmd:863]:
```suggestion
![An image in the margin](images/LinkedIn-Post-Image.png)
```

File: content/tests/styling.qmd [hunk 14]
```diff
@@ -0,0 +1,932 @@
...
+
+``` {.markdown shortcodes="false"}
+{{< var grass.version >}}
+```
+
+Will render {{< var grass.version >}}
+
+## Includes
+
+The include [shortcode](https://quarto.org/docs/authoring/includes.html) allow us to use reusable markdown.
```
Comment (veroandreo) [inline content/tests/styling.qmd:900]:
```suggestion
The include [shortcode](https://quarto.org/docs/authoring/includes.html) allows us to use reusable markdown. 
```

**Review by veroandreo — DISMISSED** [review]
I didn´t test locally as I need to update quarto and all. So, I marked typos mostly. Indeed, I also thought we agreed on sentence case for titles.

File: content/tests/styling.qmd
Comment (petrasovaa) [reply content/tests/styling.qmd:1]: I don't understand, I didn't do anything and still it shows up.

Comment (cwhite911) [reply content/tests/styling.qmd:1]: Hmm...I must have something misconfigured. The expected behavior is that those pages should only render when the env `QUARTO_PROFILE=local`.

**Review by cwhite911 — COMMENTED** [review]
I didn't know about the decision about title being sentence case. I will update the materials to reflect this.

File: content/tests/styling.qmd (hunk 12 above)
Comment (cwhite911) [reply content/tests/styling.qmd:831]: I will double check.

Comment (cwhite911) [reply content/tests/styling.qmd]: I've updated this.

File: content/tests/styling.qmd
Comment (cwhite911) [reply content/tests/styling.qmd:1]: I've fixed this issue, by setting up profiles that allow us to render a dev/local version and a production version.

https://quarto.org/docs/projects/profiles.html

File: _variables.yml (hunk 1 above)
Comment (cwhite911) [reply _variables.yml:3]: I think it will be good to have variable with the current GRASS version, so we can be version specific in the tutorials.

**Comment by petrasovaa** [comment]
It looks nice! I would hide the guide from the front page, it's not really a guide, it shows qmd syntax. Maybe we could make it into a contributing guide, or link it from README.

**Comment by petrasovaa** [comment]
BTW, there is some incosistency in headings - some of the headings (h2) are not bold in the screenshot, but they are bold when I render it locally. I like the bold ones. This needs to be corrected in the grass doc as well, they are not bold.

**Comment by cwhite911** [comment]
> It looks nice! I would hide the guide from the front page, it's not really a guide, it shows qmd syntax. Maybe we could make it into a contributing guide, or link it from README.

The guide isn't visible at the moment when running in production (aka built by gh-actions)

**Review by petrasovaa — APPROVED** [review]
